Storage/docu/docu_3_code.cc

```diff
@@ -484,18 +484,33 @@ TEST(StorageDocumentation, IfUnmodifiedSince) {
     EXPECT_EQ(200, static_cast<int>(response.code));
     ASSERT_TRUE(response.headers.Has("Last-Modified"));
     EXPECT_EQ("Fri, 01 Jul 2016 09:00:00 GMT", response.headers.Get("Last-Modified"));
+    ASSERT_TRUE(response.headers.Has("X-Current-Last-Modified"));
+    EXPECT_EQ("1467363600000000", response.headers.Get("X-Current-Last-Modified"));
   }
 
   Client updated_client1((ClientID(client1_key)));
   updated_client1.name = "Jane Doe";
 
+  // Try to modify the record with the incorrect header `X-Current-If-Unmodified-Since`.
+  {
+    const auto response = HTTP(PUT(base_url + "/api/data/client/" + client1_key_str, updated_client1)
+                                   .SetHeader("X-Current-If-Unmodified-Since", "-1"));
+    EXPECT_EQ(400, static_cast<int>(response.code));
+  }
   // Try to modify the record with the incorrect header `If-Unmodified-Since`.
   {
     const auto response = HTTP(PUT(base_url + "/api/data/client/" + client1_key_str, updated_client1)
                                    .SetHeader("If-Unmodified-Since", "Bad string"));
     EXPECT_EQ(400, static_cast<int>(response.code));
   }
 
+  // Try to modify the record with the header `X-Current-If-Unmodified-Since` set 1 us in the past.
+  {
+    const auto header_value = current::ToString(publish_time - std::chrono::microseconds(1));
+    const auto response = HTTP(PUT(base_url + "/api/data/client/" + client1_key_str, updated_client1)
+                                   .SetHeader("X-Current-If-Unmodified-Since", header_value));
+    EXPECT_EQ(412, static_cast<int>(response.code));
```

Comment:
Perhaps look at the body too? Also, is the header with the current last modified time returned along with 412?